trees/redblacktree/redblacktree.go

```diff
@@ -192,6 +203,26 @@ func (tree *Tree) Right() *Node {
 	return parent
 }
 
+// Min returns the minimum key value pair of the AVL tree
+// or nils if the tree is empty.
+func (tree *Tree) Min() (interface{}, interface{}) {
+	n := tree.Left()
+	if n == nil {
+		return nil, nil
+	}
+	return n.Key, n.Value
+}
+
+// Max returns the minimum key value pair of the AVL tree
+// or nils if the tree is empty.
+func (tree *Tree) Max() (interface{}, interface{}) {
```

Comment:
Left and right are basically non-abstracted versions. I figured that the bidiset, treeset, treemap that use the tree interface wouldn't want to know about nodes. They just care about their keys and values. Feel free to change any way you want.